pkg/streamer/file_test.go

```diff
@@ -379,8 +394,10 @@ func (t *testUtilSuite) TestRelaySubDirUpdated(c *C) {
 	wg.Add(1)
 	go func() {
 		defer wg.Done()
-		up, err2 := relaySubDirUpdated(ctx, watcherInterval, subDir, relayPaths[1], relayFiles[1], 0)
-		c.Assert(err2, IsNil)
+		relaySubDirUpdated(ctx, watcherInterval, subDir, relayPaths[1], relayFiles[1], 0, updatePathCh, errCh)
+		c.Assert(len(errCh), Equals, 0)
+		c.Assert(len(updatePathCh), Equals, 1)
+		up := <-updatePathCh
 		c.Assert(up, Equals, relayPaths[1])
 	}()
 
```

Comment:
in line 404 there's `// wait watcher started, update the file`, don't know why this is needed

Reply:
Watcher only catch update event after watch. See https://github.com/pingcap/dm/blob/master/pkg/streamer/file.go#L198-L203